static/js/services/playsound.js

```diff
@@ -156,7 +156,71 @@ define(['underscore', 'Howler', 'require'], function(_, Howler, require) {
 	window.PLAYSOUND = registry; // make available for debug.
 
 	// playSound
-	return [function() {
+	return ['translation', 'appData', 'mediaStream', function(translation, appData, mediaStream) {
+
+		/*
+		  Api
+		  ---
+		  Each sound category: join, chat, and call can be individually set with the
+		  below keys to customize if a sound is played.
+
+		  PLAY: Sounds are played for the category
+		  CALLNOSOUND: No sounds are played for the category when in a call
+		  NOSOUND: No sounds are played for the category
+		*/
+		var scope;
+		var playSoundCategories = {
+			join: 'PLAY',
+			chat: 'PLAY',
+			call: 'PLAY'
+		}
+		var playSound = {};
+		var inCall = false;
+		var playSoundOptions = {
+			PLAY: translation._("Default"),
+			CALLNOSOUND: translation._("No sounds when in call"),
+			NOSOUND: translation._("No sounds")
+		};
+		// Check each invividual sound category setting: join, chat, call with the sound names that they emit
+		// to decide whether to play them or not
+		var shouldPlaySound = function(id) {
+			var play = true;
+
+			if (playSound.join === 'NOSOUND' || inCall && playSound.join === 'CALLNOSOUND') {
+				if (id === "left" || id === "joined") {
+					play = false;
+				}
+			}
+			if (playSound.chat === 'NOSOUND' || inCall && playSound.chat === 'CALLNOSOUND') {
+				if (id === "announce") {
+					play = false;
+				}
+			}
+			if (playSound.call === 'NOSOUND' || inCall && playSound.call === 'CALLNOSOUND') {
+				if (id === "ring" || id === "end" || id === "dial" || id === "connect" || id === "prompt") {
+					play = false;
+				}
+			}
+
+			return play;
+		};
+		mediaStream.webrtc.e.on("streamadded", function() {
+			// console.log('sound start call');
+			inCall = true;
+		});
+		mediaStream.webrtc.e.on("stop", function() {
+			// console.log('sound stop call');
+			inCall = false;
+		});
+		appData.e.on("userSettingsLoaded", function(event, loadedUser, user) {
+			if (user && user.settings && user.settings.sounds) {
+				angular.extend(playSound, user.settings.sounds);
```

Comment:
This looks strange. Why whould you extends the playSound service with the user settings?